mantis/simulation/routers/solver.py

```diff
@@ -0,0 +1,205 @@
+from dataclasses import dataclass
+from abc import ABC, abstractmethod
+
+import numpy as np
+import cvxpy as cp
+from cvxpy.expressions.expression import Expression
+from cvxpy.expressions.variable import Variable
+
+from simulation.routers.data import AllData, Input, Output, read_dummy_data
+
+
+@dataclass
+class RouterAsset:
+    id: int
+    price: float
+
+
+@dataclass
+class RouterExchange(ABC):
+    assets: list[RouterAsset]
+    fix_cost: float
+    reserves: list[float]
+    fee_in: float
+    fee_out: float
+
+    @abstractmethod
+    def constraints(self, new_reserve: Expression) -> list[Expression]:
+        pass
+
+
+class ConstantProductPool(RouterExchange):
```

Comment:
so i do not mix modelling and problem input, as modelling is more like switch case. 

that is better to handle that data is not mixed with solution lib.